spring-kafka/src/main/java/org/springframework/kafka/listener/ExceptionClassifier.java

```diff
@@ -149,6 +150,16 @@ public void setClassifications(Map<Class<? extends Throwable>, Boolean> classifi
 	@SuppressWarnings("varargs")
 	public final void addNotRetryableExceptions(Class<? extends Exception>... exceptionTypes) {
 		add(false, exceptionTypes);
+		notRetryable(Arrays.stream(exceptionTypes));
+	}
+
+	/**
+	 * Subclasses can override this to receive notification of configuration of not
+	 * retryable exceptions.
+	 * @param notRetryable the not retryable exceptions.
+	 * @since 2.9.3
+	 */
+	protected void notRetryable(Stream<Class<? extends Exception>> notRetryable) {
```

Comment:
Why `Stream`?
What is wrong with a plain `Class<? extends Exception>...` propagation?

Reply:
Because `@SafeVarags` requires a `final` modifier.